skimage/util/dtype.py

```diff
@@ -54,6 +54,126 @@ def dtype_limits(image, clip_negative=False):
     return imin, imax
 
 
+def _dtype_itemsize(itemsize, *dtypes):
+    """Return first of `dtypes` with itemsize greater than `itemsize`
+
+    Parameters
+    ----------
+    itemsize: int
+        The data type object element size.
+
+    Other Parameters
+    ----------------
+    *dtypes:
+        Any Object accepted by `np.dtype` to be converted to a data
+        type object
+
+    Returns
+    -------
+    dtype: data type object
+        First of `dtypes` with itemsize greater than `itemsize`.
+
+    """
+    return next(dt for dt in dtypes if np.dtype(dt).itemsize >= itemsize)
+
+
+def _dtype_bits(kind, bits, itemsize=1):
+    """Return dtype of `kind` that can store a `bits` wide unsigned int
+
+    Parameters:
+    kind: str
+        Data type kind.
+    bits: int
+        Desired number of bits.
+    itemsize: int
+        The data type object element size.
+
+    Returns
+    -------
+    dtype: data type object
+        Data type of `kind` that can store a `bits` wide unsigned int
+
+    """
+
+    compare_bits_to = bits.__le__ if kind == 'u' else bits.__lt__
```

Comment:
I agree with @sciunto, undo this particular optimisation, as it is much less readable than the inner `compare` function. Something preferable would be to just be explicit in the loop:

```python
s = next(i for i in (itemsize,) + (2, 4, 8)
         if bits < i*8 or bits == i*8 and kind == 'u')
```

Reply:
I like this ;-) well done @jni